packages/wallet/src/KeyManagement/LedgerKeyAgent.ts

```diff
@@ -164,11 +170,68 @@ export class LedgerKeyAgent extends KeyAgentBase {
     });
   }
 
+  async signTransaction(initializedTxData: InitializeTxResult): Promise<Cardano.Signatures> {
+    try {
+      const deviceConnection = await LedgerKeyAgent.checkDeviceConnection(this.serializableData.communicationType, this.deviceConnection);
+      const dataForSigning = this.#prepareDataForSigning(initializedTxData);
+      const response = await deviceConnection.signTransaction(dataForSigning);
+      // TODO - derive witnes(es) path(s) and return Map<Ed25519PublicKey, Ed25519Signature>
+      return response.witnesses;
+    } catch (error: any) {
+      // eslint-disable-next-line unicorn/numeric-separators-style
+      if (error.code === 28169) {
+        throw new AuthenticationError('Transaction signing aborted', error);
+      }
+      throw new TransportError('Transport failed', error);
+    }
+    return response;
+  }
+
   async signBlob(): Promise<SignBlobResult> {
     throw new NotImplementedError('signBlob');
   }
 
   async exportRootPrivateKey(): Promise<Cardano.Bip32PrivateKey> {
     throw new NotImplementedError('Operation not supported!');
   }
+
+  async #prepareDataForSigning({ body, hash, inputSelection }: InitializeTxResult) {
+    // TODO - Isolate change outputs
+    // TODO - Prepare token bundle
+    const inputs = body.inputs.map((input) => ({
+      txHashHex: input.txId,
+      outputIndex: input.txIndex,
+      // TODO - Derivation Path - get from `ownSignatureKeyPaths`
```

Comment:
I think this is an odd API and I suggest setting all paths on inputs to `null`, using `ownSignatureKeyPaths` to specify `additionalWitnessPaths` in `SignTransactionRequest`.

Note that ledger will probably be able to compute that it needs stake key (`role: KeyType.Stake`) by itself, so you may have to omit it from `additionalWitnessPaths`

Reply:
I will double check but as I can see from integration tests it is required.

Reply:
@mkazlauskas here is the definition and maybe we should construct path. Additional witnesses are used for e.g. minting
```
path: [BIP32Path](https://vacuumlabs.github.io/ledgerjs-cardano-shelley/4.0.0/index.html#bip32path) | null
Describes path used for witnessing this UTxO. The API will sign transaction with this path.

Note: null indicates we don't want to sign this utxo. This is highly non-standard and the only usecase so far is pool registration as owner. We therefore don't mark it as optional so that people won't forget specifying it

Note: Device has no ability to really check whether path is correct witnessing path for this UTxO.
```

Reply:
Hm no I think it would be easier to set all paths as "additional". Mapping inputs to addresses would make it more complicated.

Reply:
ok :)